include/libtorrent/alert_manager.hpp

```diff
@@ -116,27 +131,19 @@ namespace libtorrent {
 		template <class T>
 		bool should_post() const
 		{
-			mutex::scoped_lock lock(m_mutex);
-			if (m_alerts[m_generation].size() >= m_queue_size_limit
-				* (1 + T::priority))
-			{
-				return false;
-			}
-			return (m_alert_mask & T::static_category) != 0;
+			return (m_alert_mask.load(boost::memory_order_relaxed) & T::static_category) != 0;
```

Comment:
I would not use a relaxed load here. if you change the alert mask, which alerts that are posted after that should be deterministic and reliable. a relaxed load disconnects it from any other memory updates.

Just `(m_alert_mask & T::static_category) != 0` should be fine.